diff --git a/app.js b/app.js
index 2bb942b..5caab70 100644
--- a/app.js
+++ b/app.js
@@ -12,6 +12,18 @@ app.use(express.static('client'));
 app.use(express.json())
 app.use("/form", express.static("./client/form.html"))
 
+loadJson = function(jsonpath) {
+    return JSON.parse(
+        fs.readFileSync(data_dir + jsonpath, 'utf8')
+    );
+}
+
+saveJson = function(jsonpath, value) {
+    let jsontext = JSON.stringify(value)
+    if (jsontext) {
+        fs.writeFileSync(data_dir + jsonpath, jsontext);
+    }
+}
 
 read = function(req, resp, jsonpath){
     const entities = require(data_dir + jsonpath)
@@ -25,7 +37,7 @@ read = function(req, resp, jsonpath){
 }
 
 read_entity = function(req, resp, jsonpath){
-    const entities = require(data_dir + jsonpath)
+    const entities = loadJson(jsonpath)
     let id = req.params.id
     sent = false
     for (i in entities){
@@ -42,7 +54,7 @@ read_entity = function(req, resp, jsonpath){
 
 create = function(req, resp, jsonpath, attributes){
     console.log("Attributes", attributes)
-    let entities = require(data_dir + jsonpath)
+    let entities = loadJson(jsonpath)
     id = uuidv4()
     console.log(attributes)
     if (attributes.includes("id") == false){
@@ -61,8 +73,7 @@ create = function(req, resp, jsonpath, attributes){
         }
     }
     entities.push(newEntity)
-    let entityText = JSON.stringify(entities)
-    fs.writeFileSync(data_dir + jsonpath, entityText)
+    saveJson(jsonpath, entities);
     resp.status(200).send(newEntity)
 
 }
@@ -78,7 +89,7 @@ update = function(req, resp, jsonpath){
             ignore_list.push(param)
         }
     }
-    let allEntities = require(data_dir + jsonpath)
+    let allEntities = loadJson(jsonpath)
     filteredCars = filter_entity_list(req, allEntities, ignore=ignore_list)
     let entity = filteredCars[0]
 
@@ -94,14 +105,13 @@ update = function(req, resp, jsonpath){
     
     if (allEntities.includes(entity) == false){resp.status(404)}
     allEntities.push(entity)
-    let entityText = JSON.stringify(allEntities)
-    fs.writeFileSync(data_dir + jsonpath, entityText)
+    saveJson(jsonpath, allEntities)
     resp.status(200).send(entity)
 }
 
 remove = function(req, resp, jsonpath){
     id = req.params.id
-    const allEntities = require(data_dir + jsonpath)
+    const allEntities = loadJson(jsonpath)
     filteredEntities = filter_entity_list(req, allEntities)
     console.log(filteredEntities)
     sent = false
@@ -112,8 +122,7 @@ remove = function(req, resp, jsonpath){
             console.log("REMOVING IEM WITH ID:", entity.id)
             index = filteredEntities.indexOf(entity)
             filteredEntities.splice(index, 1)
-            let entityText = JSON.stringify(filteredEntities)
-            fs.writeFileSync(data_dir + jsonpath, entityText)
+            saveJson(filteredEntities);
             resp.status(204).send()
             sent = true
             break
